x/concentrated-liquidity/position.go

```diff
@@ -621,17 +610,54 @@ func (k Keeper) RemovePositionIdToLock(ctx sdk.Context, positionId, underlyingLo
 	store.Delete(lockIdKey)
 }
 
-// positionHasUnderlyingLockInState checks if a given positionId has a corresponding lock in state.
-func (k Keeper) positionHasUnderlyingLockInState(ctx sdk.Context, positionId uint64) (bool, error) {
+// PositionHasActiveUnderlyingLock is a non mutative method that checks if a given positionId has a corresponding lock in state.
+// If it has a lock in state, checks if that lock is still active.
+// If lock is still active, returns true.
+// If lock is no longer active, returns false.
+func (k Keeper) PositionHasActiveUnderlyingLock(ctx sdk.Context, positionId uint64) (hasActiveUnderlyingLock bool, lockId uint64, err error) {
 	// Get the lock ID for the position.
-	_, err := k.GetLockIdFromPositionId(ctx, positionId)
-	if err == nil || !errors.Is(err, types.PositionIdToLockNotFoundError{PositionId: positionId}) {
-		return true, nil
+	lockId, err = k.GetLockIdFromPositionId(ctx, positionId)
+	if errors.Is(err, types.PositionIdToLockNotFoundError{PositionId: positionId}) {
+		return false, 0, nil
+	} else if err != nil {
+		return false, 0, err
+	}
+
+	// Check if the underlying lock is mature.
+	lockIsMature, err := k.isLockMature(ctx, lockId)
+	if err != nil {
+		return true, 0, err
```

Comment:
My thought process was, if we know the position at one point had a lock, and we go to query if the lock is mature and we get an error, its best to go with the more restrictive assumption (the lock is still active) as a defense in depth in case, for whatever reason, we dont stop at the error.

Reply:
https://github.com/osmosis-labs/osmosis/pull/5042/commits/df780d073fe10d10d7aab35923309b92617c6327